src/cockpit/router.py

```diff
@@ -26,12 +27,47 @@
 logger = logging.getLogger(__name__)
 
 
+class ExecutionQueue(collections.deque):
+    def run(self):
+        for method, args in self:
+            method(*args)
```

Comment:
If you try to call run() on a still frozen endpoint, it would re-queue all method calls again. Can this grow some assertion that the call does not lead to calling .append(), to avoid programming errors?